internal/transport/http2_client.go

```diff
@@ -139,15 +139,19 @@ type http2Client struct {
 }
 
 func dial(ctx context.Context, fn func(context.Context, string) (net.Conn, error), addr resolver.Address, useProxy bool, grpcUA string) (net.Conn, error) {
-	if fn != nil {
-		return fn(ctx, addr.Addr)
-	}
 	networkType := "tcp"
 	address := addr.Addr
-	if n, ok := networktype.Get(addr); ok {
+	n, ok := networktype.Get(addr)
+	if fn != nil {
+		if ok && n == "unix" {
+			return fn(ctx, "unix:///"+address)
```

Comment:
This deserves a comment.  Something like: `For backward compatibility, if the user dialed "unix:///path", the passthrough resolver would be used and the user's custom dialer would see "unix:///path".  Re-add "unix://" here since we now support the "unix" scheme by default, which strips this prefix.`

Reply:
Added this comment.